Dockerfile

```diff
@@ -16,76 +10,118 @@ ENV BUILD_ROOT /buildroot
-    make -j $(nproc) && \
-    make install-world && \
-    rm -rf /usr/local-pg14/include; else mkdir /usr/local-pg14; fi
-RUN if [ "${PGTARGET}" -gt 15 ]; then cd postgresql-15.* && \
-    ./configure --prefix=/usr/local-pg15 --with-openssl=no --without-readline --with-system-tzdata=/usr/share/zoneinfo --with-icu --with-lz4 --with-zstd --enable-debug=no CFLAGS="-Os" && \
-    make -j $(nproc) && \
-    make install-world && \
-    rm -rf /usr/local-pg15/include; else mkdir /usr/local-pg15; fi
+  ./configure --prefix=/usr/local-pg11 --with-openssl=no --without-readline --with-icu --with-system-tzdata=/usr/share/zoneinfo --enable-debug=no CFLAGS="-Os" && \
+  make -j $(nproc) && \
+  make install && \
+  rm -rf /usr/local-pg11/include
+
+### PostgreSQL 12
+FROM base-build as build-12
+RUN wget https://ftp.postgresql.org/pub/source/v12.18/postgresql-12.18.tar.bz2 && \
+  tar -xf postgresql-12*.tar.bz2
+
+RUN cd postgresql-12.* && \
+  ./configure --prefix=/usr/local-pg12 --with-openssl=no --without-readline --with-icu --with-system-tzdata=/usr/share/zoneinfo --enable-debug=no CFLAGS="-Os" && \
+  make -j $(nproc) && \
+  make install && \
+  rm -rf /usr/local-pg12/include
+
+### PostgreSQL 13
+FROM base-build as build-13
+
+RUN wget https://ftp.postgresql.org/pub/source/v13.14/postgresql-13.14.tar.bz2 && \
+  tar -xf postgresql-13*.tar.bz2
+
+RUN cd postgresql-13.* && \
+  ./configure --prefix=/usr/local-pg13 --with-openssl=no --without-readline --with-icu --with-system-tzdata=/usr/share/zoneinfo --enable-debug=no CFLAGS="-Os" && \
+  make -j $(nproc) && \
+  make install && \
+  rm -rf /usr/local-pg13/include
+
+### PostgreSQL 14
+FROM base-build as build-14
+
+RUN wget https://ftp.postgresql.org/pub/source/v14.11/postgresql-14.11.tar.bz2 && \
+  tar -xf postgresql-14*.tar.bz2
+
+RUN cd postgresql-14.* && \
+  ./configure --prefix=/usr/local-pg14 --with-openssl=no --without-readline --with-icu --with-lz4 --with-system-tzdata=/usr/share/zoneinfo --enable-debug=no CFLAGS="-Os" && \
+  make -j $(nproc) && \
+  make install && \
+  rm -rf /usr/local-pg14/include
+
+### PostgreSQL 15
+FROM base-build as build-15
+
+RUN wget https://ftp.postgresql.org/pub/source/v15.6/postgresql-15.6.tar.bz2 && \
+  tar -xf postgresql-15*.tar.bz2
+
+RUN cd postgresql-15.* && \
+  ./configure --prefix=/usr/local-pg15 --with-openssl=no --without-readline --with-icu --with-lz4 --with-system-tzdata=/usr/share/zoneinfo --enable-debug=no CFLAGS="-Os" && \
+  make -j $(nproc) && \
+  make install && \
+  rm -rf /usr/local-pg15/include
+
+### PostgreSQL 16
+FROM base-build as build-16
+
+RUN wget https://ftp.postgresql.org/pub/source/v16.2/postgresql-16.2.tar.gz && \
+  tar -xf postgresql-16*.tar.gz
+
+RUN cd postgresql-16.* && \
+  ./configure --prefix=/usr/local-pg16 --with-openssl=no --without-readline --with-icu --with-lz4 --with-system-tzdata=/usr/share/zoneinfo --enable-debug=no CFLAGS="-Os" && \
+  make -j $(nproc) && \
+  make install && \
```

Comment:
I adjusted it for all PG versions now, just to keep it consistent, in [bad495a](https://github.com/pgautoupgrade/docker-pgautoupgrade/pull/26/commits/bad495a689622cb2af820093fd60d5c8254d6598).

The build will likely take a while, since it will not be able to use the build images. good test in any case :sweat_smile: